src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -237,12 +238,29 @@ export const txHistoryApi = createApi({
         let currentCursor: string = ''
         try {
           do {
-            const { cursor, transactions } = await adapter.getTxHistory({
+            const { cursor, transactions: _transactions } = await adapter.getTxHistory({
               cursor: currentCursor,
               pubkey,
               pageSize: 100,
             })
 
+            const transactions = _transactions.filter(transaction => {
+              const transfers = transaction.transfers
+              // Not an NFT, we don't need to filter this
+              if (!transfers.some(transfer => isNft(transfer.assetId))) return true
+              if (
+                transfers.some(
+                  transfer =>
+                    nftNameBlacklistRegex.test(
+                      `${transfer.token?.name ?? ''} ${transfer.token?.symbol ?? ''}`,
+                    ) || BLACKLISTED_COLLECTION_IDS.includes(transfer.assetId),
+                )
+              )
+                return false
+
+              return true
+            })
+
```

Comment:
Move to `updateOrInsertTx` so that the logic happens as the underlying logic of both `upsertTxs` (initial fetching) and `onMessage` (incoming Tx upsertion)

Reply:
e298df99bddded0a3d9f803062649fce0c1978db